dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/filter-fee-record-payment-entity-groups.test.ts

```diff
@@ -1,301 +1,285 @@
-      );
-
-      // Assert
-      expect(result).toContainEqual(firstFeeRecordPaymentGroup);
-      expect(result).not.toContainEqual(secondFeeRecordPaymentGroup);
-      expect(result).toContainEqual(anotherFeeRecordPaymentGroup);
-    });
-  });
+  const aFeeRecordWithFacilityId = (facilityId: string) =>
+    FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withId(feeRecordIdGenerator.next().value).withFacilityId(facilityId).build();
 
-  describe('filterFeeRecordPaymentEntityGroupsByPaymentDetailsFilters', () => {
+  describe('filterFeeRecordPaymentEntityGroups', () => {
     describe('facility id filter', () => {
       it('returns the group where the supplied facility id filter matches one of the fee records in a group', () => {
         // Arrange
-        const facilityIdFilter = firstFeeRecords[0].facilityId;
-        const paymentDetailsFilters = { facilityId: facilityIdFilter };
+        const firstGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('11111111'), aFeeRecordWithFacilityId('22222222')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const secondGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('33333333')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const groups = [firstGroup, secondGroup];
+
+        const facilityIdFilter = '22222222';
+        const filters = { facilityId: facilityIdFilter };
 
         // Act
-        const result = filterFeeRecordPaymentEntityGroupsByPaymentDetailsFilters(allFeeRecordPaymentGroups, paymentDetailsFilters);
+        const result = filterFeeRecordPaymentEntityGroups(groups, filters);
 
         // Assert
-        expect(result).toContainEqual(firstFeeRecordPaymentGroup);
-        expect(result).not.toContainEqual(secondFeeRecordPaymentGroup);
+        expect(result).toContainEqual(firstGroup);
+        expect(result).not.toContainEqual(secondGroup);
       });
 
       it('returns the group where the supplied facility id filter partially matches a facility id in the group', () => {
         // Arrange
-        const facilityIdFilter = '1111'; // corresponds to firstFeeRecords[0].facilityId
-        const paymentDetailsFilters = { facilityId: facilityIdFilter };
+        const firstGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('11111111'), aFeeRecordWithFacilityId('22222222')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const secondGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('33333333')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const groups = [firstGroup, secondGroup];
+
+        const facilityIdFilter = '1111'; // corresponds to firstGroup.feeRecords[0].facilityId
```

Comment:
Can we remove the comment on this line and all similar ones now? It should be very obvious from the test set up where these values come from now they are all in the test itself.